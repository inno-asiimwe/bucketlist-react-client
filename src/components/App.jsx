import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import store from '../config/store';
import Home from './Home';
import RegisterUser from '../containers/SignupForm';
import LoginUser from '../containers/LoginForm';
import NewBucketlist from '../containers/NewBucketlist';
import EditBucketlist from '../containers/EditBucketlist';
import ShowBucketlist from '../containers/Bucketlist';
import NewItem from '../containers/NewBucketlistItem';
import EditBucketlistItem from '../containers/EditBucketlistItem';
// import Nav from './NavBar';
import Logout from '../containers/Logout';

const App = () => (
  <Provider store={store}>
    <BrowserRouter>
      <div>
        <Switch>
          <Route path="/bucketlists/:id/:itemid/edit" component={EditBucketlistItem} />
          <Route exact path="/bucketlists/new" component={NewBucketlist} />
          <Route exact path="/bucketlists/:id/edit" component={EditBucketlist} />
          <Route path="/bucketlists/:id/new" component={NewItem} />
          <Route path="/bucketlists/:id" component={ShowBucketlist} />
          <Route path="/register" component={RegisterUser} />
          <Route path="/login" component={LoginUser} />
          <Route path="/logout" component={Logout} />
          <Route path="/" component={Home} />
        </Switch>
      </div>
    </BrowserRouter>
  </Provider>
);
export default App;
