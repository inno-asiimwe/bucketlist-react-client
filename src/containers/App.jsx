import React, { Component } from 'react';


const App = () => (
  <div>
    <h3>Home</h3>
  </div>
);


export default App;
