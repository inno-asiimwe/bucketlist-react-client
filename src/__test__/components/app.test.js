import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import App from '../../components/App';

describe('App', () => {
  it('renders without crashing', () => {
    shallow(<App />);
  });
  it('matches snapshot', () => {
    const output = shallow(<App />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});

