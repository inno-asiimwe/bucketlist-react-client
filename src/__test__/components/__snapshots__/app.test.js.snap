// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <BrowserRouter>
    <div>
      <Switch>
        <Route
          component={[Function]}
          path="/bucketlists/:id/:itemid/edit"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/bucketlists/new"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/bucketlists/:id/edit"
        />
        <Route
          component={[Function]}
          path="/bucketlists/:id/new"
        />
        <Route
          component={[Function]}
          path="/bucketlists/:id"
        />
        <Route
          component={[Function]}
          path="/register"
        />
        <Route
          component={[Function]}
          path="/login"
        />
        <Route
          component={[Function]}
          path="/logout"
        />
        <Route
          component={[Function]}
          path="/"
        />
      </Switch>
    </div>
  </BrowserRouter>
</Provider>
`;
